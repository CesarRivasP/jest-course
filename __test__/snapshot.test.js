import { getCharacter } from '../snapshot';
import rick from '../rick.json';


describe("Its time for Snapshots", () => {
test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

test('Siempre fallara la instantanea', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };

    expect(getCharacter(user)).toMatchSnapshot();
  });

  test('Tenemos una excepcion de un valor dentro del codigo', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Cesar Rivas'
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    });
  });
});