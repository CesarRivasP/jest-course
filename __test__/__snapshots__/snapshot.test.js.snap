// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Its time for Snapshots Siempre fallara la instantanea 1`] = `
Object {
  "gender": undefined,
  "id": 0,
  "name": undefined,
  "species": undefined,
  "status": undefined,
}
`;

exports[`Its time for Snapshots Snapshot 1`] = `
Object {
  "gender": "Mal",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Its time for Snapshots Tenemos una excepcion de un valor dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Cesar Rivas",
}
`;
